src/texttransformation.js

```diff
@@ -92,12 +92,44 @@ export default class TextTransformation extends Plugin {
 				include: DEFAULT_TRANSFORMATIONS
 			}
 		} );
+
+		this.editor = editor;
 	}
 
 	/**
 	 * @inheritDoc
 	 */
 	init() {
+		const model = this.editor.model;
+		const modelDocument = model.document;
+		const modelSelection = modelDocument.selection;
+
+		/**
+		 * Holds a set of active {@link module:typing/textwatcher~TextWatcher}
+		 *
+		 * @type {Set.<TextWatcher>}
+		 * @private
+		 */
+		this._watchersStack = new Set();
+
+		this.on( 'change:isEnabled', () => {
```

Comment:
And I think that being enabled/disabled is also a property of a text watcher. It makes sense to enable/disable a text watcher while holding its instance. As we hae the first case others might also need this.

This way we will simplify this class by binding `watcher.bind( 'isEnabled' ).to.( this );`. But the `TextWatcher` class will need to be updated.

Reply:
Also, I think the `watcherStack` shouldn't be required then.

Reply:
Changed, hope I've done it right ;-)